### ISSUE TITLE
remove stopwords from results?

### ISSUE THREAD

**Issue by brianhance** [MaartenGr/BERTopic#181]
My current results table top hits includes many stopwords. In the examples, there is a parameter 'english' passed to remove stopwords, but there is no arguement to pass in the BERTopic version I have installed. Is there a way to filter out stopwords from results? I am using a SentenceTransformer model.

Here is my results table:
Topic | Count | Name
-- | -- | --
-1 | 168 | -1_and_to_of_in
0 | 116 | 0_to_and_is_of
1 | 74 | 1_of_data_for_to
2 | 38 | 2_invention_or_of_methods
3 | 24 | 3_layer_conductive_aluminum_alloy

Thank you in advance

**Comment by MaartenGr** [comment]
To remove stopwords, you can use the follow along with the code [here](https://maartengr.github.io/BERTopic/tutorial/models/models.html):

```python
from bertopic import BERTopic
from sklearn.feature_extraction.text import CountVectorizer

vectorizer_model = CountVectorizer(ngram_range=(1, 2), stop_words="english")
topic_model = BERTopic(vectorizer_model=vectorizer_model)
```

The `stop_words` parameter also takes in a list of words if you have a custom list of stopwords. 

Just a quick note though, it does seem that you have little data available from the size of the topics that were created which could lead to the inclusion of stopwords. Adding more data would solve this problem as it has more instances that can be used to create more accurate topic representations.

**Comment by Avb28** [comment]
Hi Maarten,

Thanks a lot for all your answers on this forum and the extensive webpages on BERTopic. They've really helped me a lot. 

I also have a question regarding the removal of stopwords. I've succesfully removed the stopwords from my topics. See my code below. 

# Import packages 
import re
import pandas as pd
import numpy as np
import spacy
from bertopic import BERTopic
from umap import UMAP
from sklearn.feature_extraction.text import CountVectorizer

# Load Spacy model for Dutch language
spacy.load('nl_core_news_lg')
import nl_core_news_lg
nlp = nl_core_news_lg.load()
stopwords = nlp.Defaults.stop_words

# Create topic model 
`vectorizer_model = CountVectorizer(ngram_range=(1, 3), stop_words=stopwords)`
`umap_model = UMAP(n_neighbors=15, n_components=5, min_dist=0.0, metric='cosine', random_state=42)`
`topic_model = BERTopic(language="multilingual",umap_model=umap_model,diversity=0,top_n_words=10,min_topic_size=10,n_gram_range=(1,3),vectorizer_model=vectorizer_model,calculate_probabilities=True,verbose=True)`
`topics, probs = topic_model.fit_transform(articles)`

I hereafter wanted to reduce the number of outliers in this model by setting the probability threshold to 0.01 

`probability_threshold = 0.01
new_topics = [np.argmax(prob) if max(prob) >= probability_threshold else -1 for prob in probs]
topic_model.update_topics(articles, new_topics)`

But in the output of this updated topic model the stopwords have returned. I cannot figure out how I can remove the stopwords again at this stage. Do you know a solution?

Thanks in advance!

**Comment by Avb28** [comment]
I've found the solution! (Which actually was quite simple)

`topic_model.update_topics(articles, new_topics, vectorizer_model=vectorizer_model)
`
